velox/substrait/SubstraitToVeloxExpr.cpp

```diff
@@ -339,11 +338,50 @@ SubstraitVeloxExprConverter::toVeloxExpr(
   return std::make_shared<core::CastTypedExpr>(type, inputs, nullOnFailure);
 }
 
-std::shared_ptr<const core::ITypedExpr>
-SubstraitVeloxExprConverter::toVeloxExpr(
+std::shared_ptr<const core::ConstantTypedExpr>
+SubstraitVeloxExprConverter::literalsToConstantExpr(
+    const std::vector<::substrait::Expression::Literal>& literals) {
+  std::vector<variant> variants;
+  variants.reserve(literals.size());
+  VELOX_CHECK_GE(literals.size(), 0, "List should contain at least one item.");
+  for (const auto& literal : literals) {
+    variants.emplace_back(toVeloxExpr(literal)->value());
+  }
+  auto varArray = variant::array(variants);
+  ArrayVectorPtr arrayVector =
+      core::variantArrayToVector(varArray.inferType(), varArray.array(), pool_);
+  // Wrap the array vector into constant vector.
+  auto constantVector =
+      BaseVector::wrapInConstant(1 /*length*/, 0 /*index*/, arrayVector);
+  return std::make_shared<const core::ConstantTypedExpr>(constantVector);
+}
+
+core::TypedExprPtr SubstraitVeloxExprConverter::toVeloxExpr(
+    const ::substrait::Expression::SingularOrList& singularOrList,
+    const RowTypePtr& inputType) {
+  VELOX_CHECK(
+      singularOrList.options_size() > 0, "At least one option is expected.");
+  auto options = singularOrList.options();
+  std::vector<::substrait::Expression::Literal> literals;
```

Comment:
Can we avoid creating this vector and create a vector of TypedExprPtr instead?

Reply:
Fixed.